packages/eslint-plugin/src/rules/jsx-pragma/index.tsx

```diff
@@ -0,0 +1,144 @@
+import type { Rule } from 'eslint';
+import { addImportToDeclaration, removeImportFromDeclaration } from '../../utils/ast-string';
+import { findCompiledImportDeclarations, findDeclarationWithImport } from '../../utils/ast';
+
+type Options = {
+  pragma: 'jsx' | 'jsxImportSource';
+};
+
+const rule: Rule.RuleModule = {
+  meta: {
+    fixable: 'code',
+    type: 'problem',
+    messages: {
+      missingPragma: 'The {{ pragma }} pragma is missing.',
+      preferJsxImportSource:
+        'Use of the jsxImportSource pragma (automatic runtime) is preferred over the jsx pragma (classic runtime).',
+      preferJsx:
+        'Use of the jsx pragma (classic runtime) is preferred over the jsx pragma (automatic runtime).',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          pragma: {
+            type: 'string',
+            pattern: '^(jsx|jsxImportSource)$',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  create(context) {
+    const options: Options = context.options[0] || { pragma: 'jsxImportSource' };
+    const source = context.getSourceCode();
+    if (source.text.indexOf('css={') === -1) {
```

Comment:
🤔 Wanted to bail early if the source code didn't include a css prop. Could there be a better way to check this?